### ISSUE TITLE
Error resolving Microsoft.AspNetCore.Mvc.Core.dll with dotnet core version 3.1.4

### ISSUE THREAD

**Issue by JasonBro** [reinforced/Reinforced.Typings#173]
Hi Pavel,

since updating to .netcore 3.1.4 I'm getting this error with Reinforced.Typings 1.5.7 :-

```
2>        Looking into Microsoft.AspNetCore.Mvc.Core
2>        Reinforced.Typings : Build warning RT0099: Assembly Microsoft.AspNetCore.Mvc.Core from C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App\3.1.3\Microsoft.AspNetCore.Mvc.Core.dll was not loaded: System.IO.FileNotFoundException: Could not load file or assembly 'C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App\3.1.3\Microsoft.AspNetCore.Mvc.Core.dll'. The system cannot find the path specified.
2>        File name: 'C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App\3.1.3\Microsoft.AspNetCore.Mvc.Core.dll'
```
This .dll exist in the following directories:-

```
C:\Program Files\dotnet\packs\Microsoft.AspNetCore.App.Ref\3.1.0\ref\netcoreapp3.1\Microsoft.AspNetCore.Mvc.Core.dll
C:\Program Files\dotnet\packs\Microsoft.AspNetCore.App.Ref\3.1.3\ref\netcoreapp3.1\Microsoft.AspNetCore.Mvc.Core.dll
C:\Program Files\dotnet\sdk\NuGetFallbackFolder\microsoft.aspnetcore.mvc.core\2.2.0\lib\netstandard2.0\Microsoft.AspNetCore.Mvc.Core.dll
C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App\2.1.18\Microsoft.AspNetCore.Mvc.Core.dll
C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App\2.2.8\Microsoft.AspNetCore.Mvc.Core.dll
C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App\3.1.0\Microsoft.AspNetCore.Mvc.Core.dll
C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App\3.1.4\Microsoft.AspNetCore.Mvc.Core.dll
```

My guess is that it's a problem converting the reference path version (which is still 3.1.3) to the actual path (which is now 3.1.4), although I've not been able to debug this further.

I've tried specifying the correct assembly path with `RtAdditionalAssembly`, but it just fails on the next .dll it encounters.

I've also tried fixing it with the `RtAssemblyReplacements` RegEx expressions, but unfortunately I've not had any success.

Is there anything I can try to help fix this?

Cheers - and thanks for an excellent library!

Jason

**Comment by JasonBro** [comment]
If anyone else has this issue, then there is a workround available by just copying the `C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App\3.1.4` directory to `C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App\3.1.3`, and that stops the error.

**Comment by pavel-b-novikov** [comment]
Or use `mklink`.
Sorry, that I cannot fix these issues, but MS flipped table over when added Ref assemblies. I did everything that I could by adding `RtAssemblyReplacements`, but seems that this issue slowly kills RT.
Sad.

**Comment by JasonBro** [comment]
I'll have a quick look at it over the weekend if I have time and see if I can find a better fix. I'll let you know how I get on...

**Comment by JasonBro** [comment]
I had a good look at this and I've not managed to find a fix for it. I think I'll stick with the workround for the moment.

**Comment by InternalUserError** [comment]
First, let me echo Jason's gratitude for this wonderful library!

I also ran into this issue after upgrading to Visual Studio 16.6. I believe this is related:
https://github.com/dotnet/aspnetcore/issues/22359#issuecomment-636076957

Based on the comment linked above I worked around this issue by adding the following to my csproj file:
`<ItemGroup>
  <KnownFrameworkReference Update="Microsoft.AspNetCore.App">
    <TargetingPackVersion>3.1.0</TargetingPackVersion>
  </KnownFrameworkReference>
<ItemGroup>
`

**Comment by JasonBro** [comment]
Yep - that worked for me as well. From the comment you mentioned, it looks like this will be fixed in the next release (3.1.301) in June 

On that note, I think we can close this.
